ratik/src/main/java/org/opencadc/ratik/ArtifactValidator.java

```diff
@@ -96,15 +120,383 @@ public ArtifactValidator(ArtifactDAO artifactDAO, URI resourceID, boolean trackS
+        //TODO do action
+    }
+
+    /**
+     * discrepancy: artifact not in L AND artifact in R
+     *
+     * @param remote                        The remote Artifact.
+     * @throws InterruptedException         Thread interrupted.
+     * @throws IOException                  For unreadable configuration files.
+     * @throws ResourceNotFoundException    For any missing required configuration that is missing.
+     * @throws TransientException           Temporary failure of TAP service: same call could work in future.
+     */
+    protected void validateRemote(Artifact remote)
+        throws InterruptedException, IOException, ResourceNotFoundException, TransientException {
+        // explanation1: deleted from L, pending/missed DeletedArtifactEvent in R
+        // evidence: DeletedArtifactEvent in L
+        // action: none
+        log.debug("checking explanation 1");
+        DeletedArtifactEvent localDeletedArtifactEvent = this.deletedArtifactEventDAO.get(remote.getID());
+        if (localDeletedArtifactEvent != null) {
+            log.info(String.format("Artifact not in L && in R - Explanation1: DeletedArtifactEvent in L\n"
+                                       + "action: no action necessary\n"
+                                       + "remote - %s\nlocal DeletedArtifactEvent - %s",
+                                   remote, localDeletedArtifactEvent));
+            return;
+        }
+
+        // explanation2: L==storage, deleted from L, pending/missed DeletedStorageLocationEvent in R
+        // evidence: DeletedStorageLocationEvent in L
+        // action: none
+        log.debug("checking explanation 2");
+        if (!this.trackSiteLocations) {
+            DeletedStorageLocationEvent localDeletedStorageLocationEvent = this.deletedStorageLocationEventDAO.get(remote.getID());
+            if (localDeletedStorageLocationEvent != null) {
+                log.info(String.format("Artifact not in L && in R - Explanation2: DeletedStorageLocationEvent in L\n"
+                                           + "action: no action necessary"
+                                           + "\nremote - %s\nlocal DeletedStorageLocationEvent - %s",
+                                       remote, localDeletedStorageLocationEvent));
+                return;
+            }
+        }
+
+        // explanation3: L==storage, new Artifact in R, pending/missed new Artifact event in L
+        // also
+        // explanation0: filter policy at L changed to include artifact in R
+        // explanation6: deleted from L, lost DeletedArtifactEvent
+        // explanation7: L==storage, deleted from L, lost DeletedStorageLocationEvent
+        // evidence: ?
+        // action: insert Artifact
+        log.debug("checking explanation 3");
+        if (!this.trackSiteLocations) {
+            Artifact local = this.artifactDAO.get(remote.getID());
```

Comment:
you already know the artifact is not in L by being in the validateRemote method, so nothing to check

Reply:
Do not check for the local artifact for that explanation.